lua/entities/gmod_wire_expression2/core/custom/prop.lua

```diff
@@ -454,23 +478,236 @@ end
+	this:SetAngleVelocityInstantaneous(Vector(velocity[1], velocity[2], velocity[3]))
+	ent:PhysWake()
+end
+
+__e2setcost(500)
+
+-- This code was leveraged from Garry's Mod. Perhaps it would be a bit cleaner with a slight rewrite.
+e2function void entity:makeStatue(enable)
+	if not ValidAction(self, this, "statue") then return end
+	if this:GetNWBool("IsStatue") and enable ~= 0 then return
+	elseif not this:GetNWBool("IsStatue") and enable == 0 then return end
+	
+	
+	local timeout = this.statueTimeout
+	if timeout and timeout.time > CurTime() then
+		self:throw("Wait before repeated statue operations", false)
+		return
+	end
+
+	local bones = this:GetPhysicsObjectCount()
+	if bones < 2 then return end
+	
+	if enable ~= 0 then
+		if this.StatueInfo then return end
+		local ply = self.player
+		
+		this.StatueInfo = {}
+
+		for bone = 1, bones - 1 do
+			local constraint = constraint.Weld(this, this, 0, bone, 0)
+
+			if constraint then
+				this.StatueInfo[ bone ] = constraint
+				ply:AddCleanup( "constraints", constraint )
+				undo.AddEntity( constraint )
+			end
+		end
+
+		this:SetNWBool( "IsStatue", true )
+
+		this.statueTimeout = { time = CurTime() + 1 }
+	else
+		if not this.StatueInfo then return end
+		local ply = self.player
+
+		for k, v in pairs( this.StatueInfo ) do
+
+			if IsValid( v ) then
+				v:Remove()
+			end
+
+		end
+
+		this:SetNWBool( "IsStatue", false )
+		this.StatueInfo = nil
+		this.statueTimeout = { time = CurTime() + 1 }
+
+	end
+end
+
+--------------------------------------------------------------------------------
+
 __e2setcost(20)
 e2function void entity:setPos(vector pos)
-	if not PropCore.ValidAction(self, this, "pos") then return end
-	PropCore.PhysManipulate(this, pos, nil, nil, nil, nil)
+	if not ValidAction(self, this, "pos") then return end
+	PhysManipulate(this, pos, nil, nil, nil, nil)
 end
 
 [deprecated]
 e2function void entity:reposition(vector pos) = e2function void entity:setPos(vector pos)
 
 e2function void entity:setAng(angle rot)
-	if not PropCore.ValidAction(self, this, "ang") then return end
-	PropCore.PhysManipulate(this, nil, rot, nil, nil, nil)
+	if not ValidAction(self, this, "ang") then return end
+	PhysManipulate(this, nil, rot, nil, nil, nil)
 end
 
 [deprecated]
 e2function void entity:rerotate(angle rot) = e2function void entity:setAng(angle rot)
 
+e2function void bone:setPos(vector pos)
+	ent, index = boneVerify(self, this)
+	if not ValidAction(self, ent, "pos", this, index) then return end
+	setPos( this, pos )
+	ent:PhysWake()
+end
+	
+e2function void bone:setAng(angle rot)
+	ent, index = boneVerify(self, this)
+	if not ValidAction(self, ent, "ang", this, index) then return end
+	setAng( this, rot )
+	ent:PhysWake()
+end
+
+__e2setcost(60)
+
+e2function void entity:ragdollFreeze(isFrozen)
+	if not ValidAction(self, this, "ragfreeze") then return end
+	local maxn = this:GetPhysicsObjectCount()-1
+	
+	for i = 0,maxn do
```

Comment:
Can inline the call instead of making a `maxn` variable. Same goes for the functions below this